### PR TITLE
refactor(x/mint): introduce distributeDeveloperRewards methods; unit tests (2/2) 

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: https://github.com/osmosis-labs/osmosis/issues/1916

Second part of the original PR: https://github.com/osmosis-labs/osmosis/pull/2026

This change _should not have_ state-breaking changes and does not fix rounding issues from #1917 yet.

## Brief Changelog
-  add new validation for:
       * developer vesting module account not having enough balance to distribute the dev rewards:
https://github.com/osmosis-labs/osmosis/blob/fadd99a21dfcf93b198cb40950cea7716cc47e78/x/mint/keeper/keeper.go#L274-L278
- Add more comments and link follow-up issues

## Testing and Verifying

This change added tests and can be verified as follows:
- `TestDistributeDeveloperRewards`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/mint/keeper/keeper.go [hunk 1]
```diff
@@ -286,29 +303,21 @@ func (k Keeper) distributeDeveloperRewards(ctx sdk.Context, totalMintedCoin sdk.
...
-				if err != nil {
-					return sdk.Int{}, err
-				}
-				// If recipient is vesting account, pay to account according to its vesting condition
-				err = k.bankKeeper.SendCoinsFromModuleToAccount(
-					ctx, types.DeveloperVestingModuleAcctName, devRewardsAddr, devRewardPortionCoins)
-				if err != nil {
-					return sdk.Int{}, err
-				}
+			devRewardsAddr, err := sdk.AccAddressFromBech32(w.Address)
```
Comment (p0mvn) [inline x/mint/keeper/keeper.go:306]: Removed `w.Address == ""` condition. The motivation is in the PR description. Diff is large due to tab changes

Comment (AlpinYukseloglu) [reply x/mint/keeper/keeper.go:306]: Doesn't the block of code link in the PR description specifically allow `w.Address` to be an empty string? i.e. the block of code only runs if it's not empty, so if it is then the check that prevents empty strings in `sdk.AccAddressFromBech32` is bypassed?

Zoomed out block from what you linked in PR desc:
https://github.com/osmosis-labs/osmosis/blob/c0052bbdaebac0ec94085e7742184ae2141706b9/x/mint/types/params.go#L224-L230

**Review by AlpinYukseloglu — CHANGES_REQUESTED** [review]
Overall clean/straightforward refactor! Left one comment to clarify something you mentioned in the PR desc. Other than that, LGTM

File: x/mint/keeper/keeper.go (hunk 1 above)
Comment (p0mvn) [reply x/mint/keeper/keeper.go:306]: Oh you are right! Good catch! I totally misread that check and assumed the opposite. I think it's best to keep the original approach and add more tests for that behavior

Comment (p0mvn) [reply x/mint/keeper/keeper.go]: fixed

**Review by ValarDragon — APPROVED** [review]
LGTM! This is state machine compatible, except in the edge case where the developer rewards doesn't have enough coins. 

Old behavior was to panic and halt the chain, this will just return an error So I guess it is technically compatible in that the old code would just never proceed, and not commit to a different state machine result. Thus I think this is good to backport as well

**Review by AlpinYukseloglu — APPROVED** [review]
Thanks for the fix – LGTM!

**Comment by p0mvn** [comment]
FYI, force pushed due to an issue with the merge. Had to reset and re-resolve
